src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs

```diff
@@ -458,12 +458,19 @@ private static ImmutableArray<ISymbol> GetSymbolsForNamespaceDeclarationNameCont
                 }
 
                 // If the thing on the left is a lambda expression, we shouldn't show anything.
+                var originalExpressionKind = originalExpression.Kind();
                 if (symbol.Kind == SymbolKind.Method &&
                     ((IMethodSymbol)symbol).MethodKind == MethodKind.AnonymousFunction)
                 {
                     return ImmutableArray<ISymbol>.Empty;
                 }
 
+                if (symbol.Kind == SymbolKind.Method &&
+                    (originalExpressionKind == SyntaxKind.IdentifierName || originalExpressionKind == SyntaxKind.GenericName))
```

Comment:
:bulb: Assuming the two-`SyntaxKind` overload is available at this location:

```
originalExpression.IsKind(SyntaxKind.IdentifierName, SyntaxKind.GenericName)
```

Also, can you include an example of why the previous code wouldn't work? I think seeing an example may help me avoid making a similar mistake in the future and having to go track it down. :smile: